CollapseLauncher/XAMLs/MainApp/Pages/Dialogs/KeyboardShortcuts.cs

```diff
@@ -472,161 +481,249 @@ private static async void Swap_Click(object sender, RoutedEventArgs e)
+
+        public static void LoadKbShortcuts()
         {
-            return !KeyList.Any(i => i.SequenceEqual(keys)) 
-                && !forbiddenKeyList.Any(i => i.SequenceEqual(keys));
-        }
-        #endregion
+            bool saveAfterLoad = false;
+            string keyListStr = GetAppConfigValue("KbShortcutList").ToString();
+            LogWriteLine("[KeyboardShortcuts] The following configuration is gonna be loaded from the .ini file:\n\r\t" + keyListStr);
 
-        #region Key Lists
-        private readonly static List<List<string>> defaultKeyList = new List<List<string>>
+            Dictionary<string, KbShortcut> resultList = new Dictionary<string, KbShortcut>();
+
+            if (keyListStr == null)
+            {
+                foreach (var entry in DefaultShortcutList)
+                {
+                    resultList.Add(entry.Key, entry.Value);
+                }
+
+                KbShortcutList = resultList;
+                return;
+            }
+
+            if (!keyListStr.Contains(':'))
+            {
+                saveAfterLoad = true;
+                LogWriteLine($"[KeyboardShortcuts::LoadKbShortcuts] Detected old KbShortcutList! Converting...\n\r\tOld Value:\"{keyListStr}\"",
+                    writeToLog: true);
+                var keyList = DefaultShortcutList.Keys;
+                int index = 0;
+                foreach (string combination in keyListStr.Split('|'))
                 {
-                    new List<string> { "Ctrl", "1 - 3" },   // Game selection
-                    new List<string> { "Shift", "1 - 6" },  // Region selection
+                    resultList.Add(keyList.ElementAt(index), KbShortcut.FromOldKeyList(combination.Split(",")));
+                    index++;
+                }
+            }
+            else
+            {
+                foreach (string combination in keyListStr.Split('|'))
+                {
+                    string[] values = combination.Split(":");
+                    resultList.Add(values[0], KbShortcut.FromCode(values[1]));
+                }
+            }
 
-                    new List<string> { "Ctrl", "Tab" },     // Keyboard shortcuts menu
-                    new List<string> { "Ctrl", "H" },       // Home page
-                    new List<string> { "Ctrl", "S" },       // Settings page
-                    new List<string> { "Ctrl", "Q" },       // Notification panel
+            var missingKeys = DefaultShortcutList.Keys.Except(resultList.Keys);
+            var deprecatedKeys = resultList.Keys.Except(DefaultShortcutList.Keys);
 
-                    new List<string> { "Shift", "X" },      // Screenshot folder
-                    new List<string> { "Shift", "F" },      // Game folder
-                    new List<string> { "Shift", "G" },      // Cache folder
-                    new List<string> { "Shift", "E" },      // Force close game
+            saveAfterLoad = saveAfterLoad || missingKeys.Any() || deprecatedKeys.Any();
```

Comment:
is there any reason why `saveAfterLoad` here gets compared for `saveAfterLoad` ?

Reply:
Because ```saveAfterLoad``` can already be true from the section where the old format for the shortcuts are converted (line 587)

Reply:
Noted! Thanks for the clarification~